nav2_smac_planner/src/node_lattice.cpp

```diff
@@ -69,6 +70,36 @@ void LatticeMotionTable::initMotionModel(
 
   // TODO(Matt) populate num_angle_quantization, size_x, min_turning_radius, trig_values,
   // all of the member variables of LatticeMotionTable
+
+  std::ifstream latticeFile(current_lattice_filepath);
```

Comment:
There's some more to be done if you look at my comment in https://github.com/ros-planning/navigation2/blob/8e837780fddab09dd6a39bcdf6f3cd31af3fd6cb/nav2_smac_planner/src/node_lattice.cpp#L66-L72. Basically we need to fill in the fields of the Lattice motion table https://github.com/ros-planning/navigation2/blob/8e837780fddab09dd6a39bcdf6f3cd31af3fd6cb/nav2_smac_planner/include/nav2_smac_planner/node_lattice.hpp#L80, other than `bin_size` which you can delete since that is not relevant to us right now. In place of `bin_size` we need to have a vector of angles, e.g. `vector<double>` of the length of the number of primitives so we can do something like: `bin_angles[theta]` to find the angle at a certain primitive number (to replace calculations like https://github.com/ros-planning/navigation2/blob/8e837780fddab09dd6a39bcdf6f3cd31af3fd6cb/nav2_smac_planner/src/node_lattice.cpp#L189)

Reply:
Done. However I did not know what type of state space to use.

Reply:
Good question, lets set it to DUBIN for now and then I have a bullet on my list of tasks to figure out a way to set params for reversing. We can expand that then. Add a TODO for that in the code as a reminder